forms.php

```diff
@@ -0,0 +1,84 @@
+<?php 
+    $valid=array();
+    $error=array();
+    if ($_POST) {
+        $title=$_POST['title'];
+        if (!$title)
+        {
+            $errors['title']="Името е задължително поле";
+        }
+        elseif(strlen('title')>150)
+        {
+            $errors['title']="Името трябва да бъде не повече от 150 символа";
+        }
+        else
+        {
+            $valid['title']=$title;
+        }
+        
+        $name=$_POST['name'];
+        if (!$name)
+        {
+            $errors['name']='Името е задължително поле';
+        }
+        elseif(strlen($name)>200)
+        {
+            $errors['name']="Името трябва да бъде не повече от 200 символа";
+
+        }
+        else
+        {
+            $valid['name']=$name;
+        }
+
+        $description=$_POST['description'];
+        if (!$description)
+        {
+            $errors['description']='Описанието е задължително поле';
+        }
+        elseif(strlen($description)<10)
+        {
+            $errors['description']="Описанието трябва да бъде не по-малко от 10 символа";
+
+        }
+        else
+        {
+            $valid['description']=$description;
+        }
+
+        $group=$_POST['group'];
+        if (!$description)
+        {
+            $errors['group']='Групата е задължително поле';
+        }
+        else
+        {
+            $valid['group']=$group;
+        }
+
+        $credits=$_POST['credits'];
+        if (!$description)
+        {
+            $errors['credits']='Групата е задължително поле';
+        }
+        else
+        {
+            $valid['credits']=$credits;
+        }
+    }
+        if(isset($_POST['submit']))
+        {
+            $courseTitle = $_POST['title'];
+            $name = $_POST['name'];
+            $description = $_POST['description'];
+            $group=$_POST['group'];
+            $credits = $_POST['credits'];
+            $text = $courseTitle . ", " . $name . "," . $description . ", " . $group . ", " . $credits . "\n";
+            $fp = fopen('data.txt', 'a+');
+             if(fwrite($fp, $text)) 
+             {
+                     echo 'saved';
```

Comment:
Good job for providing some information to the user about the result of the operation!

Reply:
But what about the case when there are errors in the validation? Why the saving of information is not prevented? Why the user is not informed about what is wrong and how he can fix that?
How can I (as a dumb user not knowing about the "back" button of the browser) navigate to the previous page and try to fill the form once again? What was the exact information that was recorded by the server to the file?
Here is the acceptance criteria for the task: https://iliasky.com/www/presentations/Forms.html?read=1#11